app/controllers/application_controller.rb

```diff
@@ -450,6 +450,12 @@ def redirect_back_or_default(default, escape = true, use_escaped = true)
       begin
         uri = URI.parse(escaped_back_url)
 
+        # Remove basic auth credentials
+        uri.userinfo = nil
```

Comment:
I didn't think about installations not having SSL enabled. I always assume that SSL is active but that of course is wishful thinking. So I am OK with removing the basic auth credentials (even though the attacker could have gotten the credentials on the request easily)

Plus I didn't know that the browser caches basic auth (OMG). I learned that from you, thanks.

Spec is still necessary, then.